File: backend/internal/dto/task.go
package dto

type CreateDemoTaskResult struct {
	ID     int64  `json:"id"`
	Type   string `json:"type"`
	Status string `json:"status"`
}

type ListTasksQuery struct {
	Page int `form:"page"`
	Size int `form:"size"`
}

type TaskLog struct {
	ID        int64  `json:"id"`
	Level     string `json:"level"`
	Message   string `json:"message"`
	Metadata  string `json:"metadata"`
	CreatedAt string `json:"created_at"`
}

type TaskSummary struct {
	ID          int64  `json:"id"`
	Type        string `json:"type"`
	Status      string `json:"status"`
	Progress    int    `json:"progress"`
	Error       string `json:"error_message"`
	TriggeredBy *int64 `json:"triggered_by"`
	CreatedAt   string `json:"created_at"`
	UpdatedAt   string `json:"updated_at"`
}

type ListTasksResult struct {
	Page  int           `json:"page"`
	Size  int           `json:"size"`
	Total int64         `json:"total"`
	Items []TaskSummary `json:"items"`
}

type TaskDetail struct {
	Summary TaskSummary `json:"summary"`
	Logs    []TaskLog   `json:"logs"`
}

File: frontend/src/types/task.ts
export interface TaskSummary {
  id: number;
  type: string;
  status: string;
  progress: number;
  error_message: string;
  triggered_by: number | null;
  created_at: string;
  updated_at: string;
}

export interface TaskLog {
  id: number;
  level: string;
  message: string;
  metadata: string;
  created_at: string;
}

export interface ListTasksResult {
  page: number;
  size: number;
  total: number;
  items: TaskSummary[];
}

export interface TaskDetailResult {
  summary: TaskSummary;
  logs: TaskLog[];
}

export interface CreateTaskResult {
  id: number;
  type: string;
  status: string;
}

export interface TaskStatusResult {
  id: number;
  status: string;
}
